futures-util/src/future/maybe_done.rs

```diff
@@ -44,8 +44,24 @@ pub fn maybe_done<Fut: Future>(future: Fut) -> MaybeDone<Fut> {
 }
 
 impl<Fut: Future> MaybeDone<Fut> {
+    /// Returns an optional mutable reference to the output of the future.
+    /// The output of this method will be `Some` if and only if the inner
+    /// future has been completed and [`take_output`](MaybeDone::take_output)
+    /// has not yet been called.
+    #[inline]
+    pub fn output_mut<'a>(self: PinMut<'a, Self>) -> Option<&'a mut Fut::Output> {
```

Comment:
Needs `#[allow(needless_lifetimes)] // https://github.com/rust-lang/rust/issues/52675` to make Clippy happy.

And maybe use an in-band lifetime

Reply:
(I’m sure I saw a post about potentially dropping in-band lifetimes by @aturon in the last few days, but can’t find it now)